packages/design-system/src/components/text-field.tsx

```diff
@@ -129,19 +141,72 @@ export const TextField = styled("input", {
         },
       },
     },
-    cursor: {
-      default: {
-        cursor: "default",
-        "&:focus": {
-          cursor: "text",
-        },
-      },
-      text: {
-        cursor: "text",
-      },
-    },
   },
   defaultVariants: {
     size: "1",
   },
 });
+
+type TextFieldProps = Omit<
+  React.ComponentProps<typeof TextFieldBase> & React.ComponentProps<"input">,
+  "prefix"
+> & {
+  inputRef?: React.Ref<HTMLInputElement>;
+  prefix?: React.ReactNode;
+  suffix?: React.ReactNode;
+};
+
+export const TextField = React.forwardRef<
+  React.ElementRef<typeof TextFieldBase>,
+  TextFieldProps
+>((props, forwardedRef) => {
+  const {
+    prefix,
+    suffix,
+    css,
+    disabled,
+    inputRef,
+    size,
+    state,
+    variant,
+    ...textFieldProps
+  } = props;
+  return (
+    <TextFieldBase
+      aria-disabled={disabled}
+      ref={forwardedRef}
+      size={size}
+      state={state}
+      variant={variant}
+      css={css}
+    >
+      {prefix && (
+        <Flex
+          css={{
+            alignItems: "center",
+            flexShrink: 0,
+            order: 0,
+          }}
+        >
+          {prefix}
+        </Flex>
+      )}
+
+      <InputBase ref={inputRef} disabled={disabled} {...textFieldProps} />
```

Comment:
@Andarist can you please take a look at why those two refs types are not compatible? Thanks!